File: RDD_Scala/WordLength.scala
import org.apache.spark.SparkContext
import org.apache.spark.SparkConf

val conf = new SparkConf().setAppName("WordLength").setMaster("local")
val sc = new SparkContext(conf)

val text = sc.textFile("C:/Spark/RDD_Scala/Input/WordLength.txt")

val wordLengths = text.flatMap(line => {
    val words = line.split(" ")
    words.map(word => {
        val length = word.length
        val category = length match {
            case 1 => "tiny"
            case x if x >= 1 && x <= 4 => "small"
            case x if x >= 5 && x <= 9 => "medium"
            case _ => "big"
        }
        (word, category)
    })
})

// Sắp xếp kết quả theo key (từ) theo thứ tự từ điển
val sortedCounts = wordLengths.sortByKey()
sortedCounts.saveAsTextFile("C:/Spark/RDD_Scala/Output/WordLength")
sortedCounts.foreach { case (word, category) =>
  println(s"$word: $category")}
sc.stop()

File: RDD_Python/WordLength.py
import findspark
findspark.init()

from pyspark import SparkConf, SparkContext

# Khởi tạo SparkConf
conf = SparkConf()
#conf.setMaster('spark://192.168.1.91:7077')
#conf.setMaster('local')
conf.setAppName("WordLength")

# Khởi tạo SparkContext
sc = SparkContext(conf=conf)

# Đọc dữ liệu từ tệp văn bản
text = sc.textFile("C:/Spark/RDD_Python/Input/WordLength.txt")

# Hàm phân loại từ thành phân loại tương ứng
def categorize_word(word):
    length = len(word)
    if length == 1:
        return "tiny"
    elif 1 < length <= 4:
        return "small"
    elif 5 <= length <= 9:
        return "medium"
    else:
        return "big"

# Tạo RDD mới để đếm số lượng từ trong mỗi phân loại
category_counts = text.flatMap(lambda line: line.split()) \
                      .map(lambda word: (categorize_word(word), 1)) \
                      .reduceByKey(lambda a, b: a + b)

# Lưu kết quả vào file văn bản
sorted_counts=category_counts.saveAsTextFile("C:/Spark/RDD_Python/Output/WordLength")
output = sorted_counts.collect()
for (word, count) in output:
    print("%s: %i" % (word, count))


# Dừng SparkContext
sc.stop()